src/main/java/fish/payara/extras/diagnostics/util/DomainXmlUtil.java

```diff
@@ -0,0 +1,118 @@
+
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.transform.OutputKeys;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
+import java.io.File;
+import java.util.logging.Logger;
+
+public class DomainXmlUtil {
+    private static final Logger LOGGER = Logger.getLogger(DomainXmlUtil.class.getName());
+    private static final String PASSWORD_CHANGE = "PASSWORD_HIDDEN";
+    private static final String PASSWORD_KEYWORD = "password";
+    private static final String ADMIN_PASSWORD_KEYWORD = "admin-password";
+    private static final String SEARCH_BIND_KEYWORD = "search-bind-password";
+    private static final String NAME_KEYWORD = "name";
+    private static final String VALUE_KEYWORD = "value";
+
+    public void obfuscateDomainXml (File xmlFile) {
+        try {
+            LOGGER.info("Obfuscating " + xmlFile.getAbsolutePath());
+
+            DocumentBuilderFactory dbFactory = DocumentBuilderFactory.newInstance();
+            DocumentBuilder dBuilder = dbFactory.newDocumentBuilder();
+            Document doc = dBuilder.parse(xmlFile);
+            doc.getDocumentElement().normalize();
+
+            traverseNodes(doc.getDocumentElement());
+
+            TransformerFactory transformerFactory = TransformerFactory.newInstance();
+            Transformer transformer = transformerFactory.newTransformer();
+            transformer.setOutputProperty(OutputKeys.INDENT, "yes");
+            DOMSource source = new DOMSource(doc);
+            StreamResult result = new StreamResult(xmlFile);
+            transformer.transform(source, result);
+
+            LOGGER.info("Successfully obfuscated " + xmlFile.getAbsolutePath());
+        } catch (Exception e) {
+            LOGGER.severe("Error obfuscating XML: " + e.getMessage());
+            e.printStackTrace();
+        }
+    }
+
+    private void traverseNodes(Node node) {
+        if (node.getNodeType() == Node.ELEMENT_NODE) {
+            Element tempNode = (Element) node;
+            boolean hasPasswordAttribute = tempNode.hasAttribute(PASSWORD_KEYWORD);
+            boolean hasAdminPasswordAttribute = tempNode.hasAttribute(ADMIN_PASSWORD_KEYWORD);
+            if (hasPasswordAttribute) {
+                tempNode.setAttribute(PASSWORD_KEYWORD, PASSWORD_CHANGE);
+            }
+            if (hasAdminPasswordAttribute) {
+                tempNode.setAttribute(ADMIN_PASSWORD_KEYWORD, PASSWORD_CHANGE);
+            }
+
+            String nameAttribute = tempNode.getAttribute(NAME_KEYWORD);
+            boolean hasValueAttribute = tempNode.hasAttribute(VALUE_KEYWORD);
+            if (PASSWORD_KEYWORD.equalsIgnoreCase(nameAttribute) || SEARCH_BIND_KEYWORD.equalsIgnoreCase(nameAttribute)) {
```

Comment:
Is this not missing ADMIN_PASSWORD_KEYWORD?

Reply:
No this is for passwords that are within properties so like name="Password" value="password__"

Reply:
Use regex to cover multiple scenarios like custom properties in a custom auth config, example:
````
        String regex = "[^\"]*password[^\"]*";
        Pattern pattern = Pattern.compile(regex, Pattern.CASE_INSENSITIVE);
        Matcher matcher = pattern.matcher(text);
        if (matcher.find()) {
            System.out.println(matcher.group());
        }
````

Reply:
I used it for the name using the `.contains()`. I have added the scenarios to the description.